src/utils/request.ts

```diff
@@ -117,10 +129,56 @@ async function request(config: Config): Promise<OSDataHubResponse> {
     featureCount = output.results.length;
   }
 
-  logEndConditions(config);
+  logEndConditions(config, featureCount);
+
   if (typeof output === "undefined") {
     throw Error("There is no output at the end of request");
-  } else {
-    return output;
   }
+
+  return output;
+}
+
+async function requestNGD(config: Config): Promise<FeatureCollection> {
+  let endpoint: string;
+  let featureCount = 0;
+  let output: FeatureCollection | undefined = {
+    type: "FeatureCollection",
+    features: [],
+  };
+
```

Comment:
If we need an NGD-specific request function, this should be in its own file bundled with the rest of the ngd specific code, not in a shared utils folder

Reply:
moved to ngd sub-directory